### PR TITLE
Add support for decorator options in presets

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Depends on https://github.com/babel/babel/pull/8649

Preview: https://deploy-preview-1816--babel.netlify.com/repl#?build=master

### CODE REVIEW

**Comment by littledan** [comment]
I haven't reviewed the implementation in the website itself, but it seems like a good idea to me to add these two flags, decoratorsLegacy and decoratorsBeforeExport, to the website. The descriptions, "Legacy decorators" and "Decorators before `export`" look good to me, as do the default settings and graying out of the second option when the first is set. These options are taking a pretty visually prominent place in the UI, but it seems useful to me, given that these are some of the biggest unresolved language design issues going on in TC39 right now.

File: js/repl/compile.js [hunk 1]
```diff
@@ -107,21 +107,36 @@ export default function compile(code: string, config: CompileConfig): Return {
...
           Babel.version[0] === "7" &&
           typeof preset === "string" &&
           /^stage-[0-2]$/.test(preset)
         ) {
+          const decoratorsLegacy = presetsOptions.decoratorsLegacy;
+          const decoratorsBeforeExport = decoratorsLegacy
+            ? undefined
+            : presetsOptions.decoratorsBeforeExport;
+
+          console.log("P", [
```
Comment (nicolo-ribaudo) [inline js/repl/compile.js]: TODO: delete

File: js/repl/compile.js [hunk 2]
```diff
@@ -107,21 +107,36 @@ export default function compile(code: string, config: CompileConfig): Return {
...
             preset,
             {
-              decoratorsLegacy: true,
+              decoratorsLegacy,
+              decoratorsBeforeExport,
               pipelineProposal: "minimal",
             },
           ];
         }
+        console.log("P", preset);
```
Comment (nicolo-ribaudo) [inline js/repl/compile.js]: TODO: delete

**Comment by nicolo-ribaudo** [comment]
I updated this PR; now the options are only showed when needed. Preview: deploy-preview-1816--babel.netlify.com/repl#?build=master

**Comment by existentialism** [comment]
Longstanding task to revisit the sidebar design, but functionally this LGTM!

**Pull request merged**